lib/libvarnish/vcli_serve.c

```diff
@@ -428,6 +440,12 @@ VCLS_AddFd(struct VCLS *cs, int fdi, int fdo, cls_cb_f *closefunc, void *priv)
 	cfd->cli->vlu = VLU_New(cfd, cls_vlu, *cs->maxlen);
 	cfd->cli->sb = VSB_new_auto();
 	cfd->cli->limit = cs->limit;
+	if (cs->ws_sz > 0) {
+		cfd->cli->ws = malloc(sizeof(*cfd->cli->ws));
```

Comment:
Yes, ideally we should adjust the ws to changing parameters. But I wonder how much sense this makes if we don't even want a specific parameter to tune. @bsdphk 's words on irc went like "we should have a workspace but not guaranteeing a specific size".